File: Day 1/Javascript/Day 1.js
// Advent of Code 2020
// Day 1

// Open the given input page at https://adventofcode.com/2020/day/1/input and run these scripts in dev tools console

// Part1

(() => {
    const n = document.body.innerText.split('\n').map(Number),
        l = n.length;
    for(let i = 0; i < l; ++i) {
        for(let j = 0; j < l; ++j) {
                if(n[j] + n[i] === 2020) return (n[j] * n[i])
            }
        }
    }
)();


// Part 2
(() => {
    const n = document.body.innerText.split('\n').map(Number),
        l = n.length;
    for(let i = 0; i < l; ++i) {
        for(let j = 0; j < l; ++j) {
            for(let k = 0; k < l; ++k) {
                if(n[k] + n[j] + n[i] === 2020) return (n[k] * n[j] * n[i])
            }
        }
    }
})();

File: Day 1/Python/Day 1.py
# Advent of Code 2020 Day 1

f = open("num_list.txt")
x = []
for i in f:
    x.append((int)(i.strip("\n")))
f.close()

# Part 1
for a in x:
    for b in x:
        if a + b == 2020:
            ansA = (a * b)
print(ansA)

# Part 2
for c in x:
    for d in x:
        for e in x:
            if c + d + e == 2020:
                ansB = (c * d * e)
print(ansB)
